src/app/clusters/operational-credentials-server/operational-credentials-server.cpp

```diff
@@ -70,12 +70,34 @@ class OperationalCredentialsAttrAccess : public AttributeAccessInterface
     CHIP_ERROR Read(const ConcreteReadAttributePath & aPath, AttributeValueEncoder & aEncoder) override;
 
 private:
+    CHIP_ERROR ReadNOCs(EndpointId endpoint, AttributeValueEncoder & aEncoder);
     CHIP_ERROR ReadSupportedFabrics(EndpointId endpoint, AttributeValueEncoder & aEncoder);
     CHIP_ERROR ReadCommissionedFabrics(EndpointId endpoint, AttributeValueEncoder & aEncoder);
     CHIP_ERROR ReadFabricsList(EndpointId endpoint, AttributeValueEncoder & aEncoder);
     CHIP_ERROR ReadRootCertificates(EndpointId endpoint, AttributeValueEncoder & aEncoder);
 };
 
+CHIP_ERROR OperationalCredentialsAttrAccess::ReadNOCs(EndpointId endpoint, AttributeValueEncoder & aEncoder)
+{
+    return aEncoder.EncodeList([](const auto & encoder) -> CHIP_ERROR {
+        for (auto & fabricInfo : Server::GetInstance().GetFabricTable())
+        {
+            if (!fabricInfo.IsInitialized())
+                continue;
+
+            Clusters::OperationalCredentials::Structs::NOCStruct::Type noc;
+
+            noc.fabricIndex = fabricInfo.GetFabricIndex();
+
+            ReturnErrorOnFailure(fabricInfo.GetNOCCert(noc.noc));
+            ReturnErrorOnFailure(fabricInfo.GetICACert(noc.icac));
```

Comment:
These need to be left out if the fabric index does not match the accessing fabric index.  Otherwise we are leaking certs across fabrics.

Reply:
I've filed an issue (#13594) to get this properly fixed in the cluster object layer. Perhaps we get this PR in, and correctly prioritize that bit of work?

Reply:
I would really rather prefer to not have even a temporary security bug in the tree.  Once the right fix is in place, we can remove the check here.